ush/python/pygfs/task/aero_prepobs.py

```diff
@@ -0,0 +1,237 @@
+
+import os
+import glob
+import gzip
+import tarfile
+import re
+from logging import getLogger
+from typing import List, Dict, Any, Union
+
+from wxflow import (AttrDict, FileHandler, rm_p, rmdir,
+                    Task, add_to_datetime, to_timedelta, to_datetime,
+                    datetime_to_YMD,
+                    chdir, Executable, WorkflowException,
+                    parse_j2yaml, save_as_yaml, logit)
+
+logger = getLogger(__name__.split('.')[-1])
+
+
+class AerosolObsPrep(Task):
+    """
+    Class for preparing and managing aerosol observations
+    """
+    def __init__(self, config: Dict[str, Any]) -> None:
+        super().__init__(config)
+
+        _window_begin = add_to_datetime(self.runtime_config.current_cycle, -to_timedelta(f"{self.config['assim_freq']}H") / 2)
+        _window_end = add_to_datetime(self.runtime_config.current_cycle, +to_timedelta(f"{self.config['assim_freq']}H") / 2)
+
+        local_dict = AttrDict(
+            {
+                'window_begin': _window_begin,
+                'window_end': _window_end,
+                'sensors': str(self.config['SENSORS']).split(','),
+                'data_dir': self.config['VIIRS_DATA_DIR'],
+                'input_files': '',
+                'OPREFIX': f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z.",
+                'APREFIX': f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z."
+            }
+        )
+
+        # task_config is everything that this task should need
+        self.task_config = AttrDict(**self.config, **self.runtime_config, **local_dict)
+
+    @logit(logger)
+    def initialize(self) -> None:
+        """
+        List needed raw obs files.
+        Copy the raw obs files to $DATA/obs.
+        Link over the needed executable.
+        Generate corresponding YMAL file.
+        Run IODA converter.
+        """
+        self.task_config.DATA_OBS = os.path.join(self.task_config.DATA, 'obs')
+        if os.path.exists(self.task_config.DATA_OBS):
+            rmdir(self.task_config.DATA_OBS)
+        FileHandler({'mkdir': [self.task_config.DATA_OBS]}).sync()
+
+        self.task_config.prepaero_yaml = []
+        for sensor in self.task_config.sensors:
+            raw_files = self.list_raw_files(sensor)
+            self.task_config.input_files = self.copy_obs(raw_files)
+            self.link_obsconvexe()
+            self.task_config.prepaero_config = self.get_obsproc_config(sensor)
+
+            # generate converter YAML file
+            template = f"{self.runtime_config.CDUMP}.t{self.runtime_config['cyc']:02d}z.prepaero_viirs_{sensor}.yaml"
+            _prepaero_yaml = os.path.join(self.runtime_config.DATA, template)
+            self.task_config.prepaero_yaml.append(_prepaero_yaml)
+            logger.debug(f"Generate PrepAeroObs YAML file: {_prepaero_yaml}")
+            save_as_yaml(self.task_config.prepaero_config, _prepaero_yaml)
+            logger.info(f"Wrote PrepAeroObs YAML to: {_prepaero_yaml}")
+
```

Comment:
The yaml file creation could be moved out of the loop to speed things up a tad and reduce log messages:
```suggestion
        for sensor in self.task_config.sensors:
            raw_files = self.list_raw_files(sensor)
            self.task_config.input_files = self.copy_obs(raw_files)
            self.link_obsconvexe()
            self.task_config.prepaero_config = self.get_obsproc_config(sensor)

            # generate converter YAML file
            template = f"{self.runtime_config.CDUMP}.t{self.runtime_config['cyc']:02d}z.prepaero_viirs_{sensor}.yaml"
            self.task_config.prepaero_yaml.append(_prepaero_yaml)

        _prepaero_yaml = os.path.join(self.runtime_config.DATA, template)            
        logger.debug(f"Generate PrepAeroObs YAML file: {_prepaero_yaml}")
        save_as_yaml(self.task_config.prepaero_config, _prepaero_yaml)
        logger.info(f"Wrote PrepAeroObs YAML to: {_prepaero_yaml}")

```

Reply:
I see now that this is several different yamls.  Disregard.